### DISCUSSION TITLE
Can't find variable: EventSource React Native Expo

### DISCUSSION THREAD

**Discussion by tiagodgy** [pocketbase/pocketbase#1176] [Q&A] [answered]
Trying to use Realtime i get this error:

> ReferenceError: Can't find variable: EventSource

To solve this documentations suggests:

>  npm install eventsource --save
> global.EventSource = require('eventsource');

but after install i get this error:

> The package at "node_modules\eventsource\lib\eventsource.js" attempted to import the Node standard library module "https".

This happned for different node modules, but i could fix with npm install --node_module_name, but this is not possible for http, https. So i don't know how to fix. 

Anyone using realtime in expo can give me a help?

**Comment by ganigeorgiev** [comment, marked as answer]
Hm, the `EventSource` polyfill [doesn't seem to support react native](https://github.com/EventSource/eventsource/issues/132).

You can try with different polyfill. I don't have much experience with react native so I'm not sure whether there is a good alternative, but you could give https://www.npmjs.com/package/react-native-sse a try (it seems the most up-to-date one).
Note that you still have to assign it to the global context so that JS SDK could use it, eg. something like:
```js
import EventSource from "react-native-sse";

// or globalThis depending which is available
global.EventSource = EventSource;
```

_Sometime in the future, I'm planning a rewrite of the JS SDK `RealtimeService` and the `EventSource` dependency no longer will be needed._

**Reply by tiagodgy** [reply]
Thanks. I tried it yesterday but was not able to make it work. I was using global.EventSource = EventSource**()**; instead of the way you suggested.

**Comment by anhtuank7c** [comment]
This solution works for me. Thanks @ganigeorgiev 

```
import EventSource from "react-native-sse";

// or globalThis depending which is available
global.EventSource = EventSource;
```

![image](https://user-images.githubusercontent.com/3163410/214323400-b30fd8cb-1343-4243-ac99-02ae2539b982.png)
